File: docs/lib/catalogue_filters.js
class CatalogueFilters {

    /** @type {number} */
    pageIndex;

    /** @type {number} */
    pageSize;

    /** @type {string[]} */
    ids;

    /** @type {string} */
    hint;

    constructor(json) {
        this.pageIndex = json?.pageIndex ?? null;
        this.pageSize = json?.pageSize ?? null;
        this.ids = json?.ids ?? null;
        this.hint = json?.hint ?? null;
    }

}

export {
    CatalogueFilters
}

File: Core/CatalogueFilters.cs
using KolibSoft.Catalogue.Core.Abstractions;

namespace KolibSoft.Catalogue.Core;

public class CatalogueFilters : IPageFilters, IItemFilters
{
    public int? PageIndex { get; init; }
    public int? PageSize { get; init; }
    public Guid[]? Ids { get; init; }
    public DateTime? ChangesAt { get; init; }
    public string? Hint { get; init; }
}